packages/react-query/src/useQueries.ts

```diff
@@ -188,5 +190,35 @@ export function useQueries<T extends any[]>({
     observer.setQueries(defaultedQueries, { listeners: false })
   }, [defaultedQueries, observer])
 
+  const errorResetBoundary = useQueryErrorResetBoundary()
+
+  defaultedQueries.forEach((query) => {
+    if (query.suspense || query.useErrorBoundary) {
+      // Prevent retrying failed query if the error boundary has not been reset yet
+      if (!errorResetBoundary.isReset()) {
+        query.retryOnMount = false
+      }
```

Comment:
could this be a single, abstracted function used in useBaseQuery as well? Something like:

(I'm using options instead of query already, see my previous comment)

```
ensurePreventErrorBoundaryRetry(options, errorResetBoundary)
```

Reply:
[[7f81e88](https://github.com/TanStack/query/pull/4177/commits/7f81e8893bd674f103d6b72a93b77151abea86a3)](https://github.com/TanStack/query/pull/4177/commits/7f81e8893bd674f103d6b72a93b77151abea86a3#diff-b44eb6740558a471060188bab4175b44802e0b0d56d4be245cc87acc748e44fdR11)